src/gpodder/deviceplaylist.py

```diff
@@ -104,7 +105,15 @@ def get_absolute_filename_for_playlist(self, episode):
         if foldername:
             filename = os.path.join(foldername, filename)
         if self._config.device_sync.playlists.use_absolute_path:
-            filename = os.path.join(util.relpath(self._config.device_sync.device_folder, self.mountpoint.get_uri()), filename)
+            # find mount point, ensuring we don't end up locked up in the loop
+            drive_start = self._config.device_sync.device_folder
+            while os.path.ismount(drive_start) is not True and drive_start != os.path.dirname(drive_start):
+                drive_start = os.path.dirname(drive_start)
+            filename = os.path.join(util.relpath(self._config.device_sync.device_folder, drive_start), filename)
+            # distinguish this as an absolute path relative to the device
+            filename = "/" + filename
```

Comment:
is the rest of the code supposed to work with url as a path (eg `mtp:///`)?
Then prepending a slash will break it: `/mtp:///`

Reply:
The intention is to not have a prefix like mtp:// or file:// at all for the filename.

I actually had to adjust things because self.mountpoint.get_uri() was giving me "file://path/to/somewhere". I _think_ this will tend to give just a regular file path, but I'm not knowledgeable enough to know if it can be guaranteed or not with the way it is right now. It does work for me as is right now, at least.

Reply:
Perhaps I should check if the filename is a url or a regular file path and try to strip any prefix we find? I think that should be do-able.

Reply:
I believe this is sufficiently handled with the newest commit, [f475386](https://github.com/gpodder/gpodder/pull/1672/commits/f47538601754299b3bb8c308c75bac81f0491789). Purposefully strip file: and mtp: prefixes for m3u entry paths.